bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikRouterosBridgeHandler.java

```diff
@@ -0,0 +1,271 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mikrotik.internal.handler;
+
+import static org.openhab.binding.mikrotik.internal.MikrotikBindingConstants.*;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mikrotik.internal.MikrotikBindingConstants;
+import org.openhab.binding.mikrotik.internal.config.RouterosThingConfig;
+import org.openhab.binding.mikrotik.internal.model.RouterosDevice;
+import org.openhab.binding.mikrotik.internal.model.RouterosRouterboardInfo;
+import org.openhab.binding.mikrotik.internal.model.RouterosSystemResources;
+import org.openhab.binding.mikrotik.internal.util.StateUtil;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ThingStatusInfoBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import me.legrange.mikrotik.MikrotikApiException;
+
+/**
+ * The {@link MikrotikRouterosBridgeHandler} is a main binding class that wraps a {@link RouterosDevice} and
+ * manages fetching data from RouterOS. It is also responsible for updating brindge thing properties and
+ * handling commands, which are sent to one of the channels and emit channel updates whenever required.
+ *
+ * @author Oleg Vivtash - Initial contribution
+ */
+@NonNullByDefault
+public class MikrotikRouterosBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(MikrotikRouterosBridgeHandler.class);
+    private @Nullable RouterosThingConfig config;
+    private @Nullable volatile RouterosDevice routeros;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private Map<String, State> currentState = new HashMap<>();
+
+    public static boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return MikrotikBindingConstants.THING_TYPE_ROUTEROS.equals(thingTypeUID);
+    }
+
+    public MikrotikRouterosBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        cancelRefreshJob();
+        config = getConfigAs(RouterosThingConfig.class);
+
+        logger.debug("Initializing MikrotikRouterosBridgeHandler with config = {}", config);
+        if (config.isValid()) {
+            routeros = new RouterosDevice(config.host, config.port, config.login, config.password);
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, String.format("Connecting to %s", config.host));
```

Comment:
The UNKNOWN state has no details. See https://www.openhab.org/docs/concepts/things.html#status-details